lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -381,11 +378,19 @@ private JourneyResponse buildF2FNoMatchResponse(
     }
 
     private JourneyResponse buildNoMatchResponse(
-            List<VcStoreItem> vcStoreItems, AuditEventUser auditEventUser) throws SqsException {
+            List<VcStoreItem> vcStoreItems,
+            AuditEventUser auditEventUser,
+            ContraIndicators contraIndicators)
+            throws SqsException, MitigationRouteConfigNotFoundException, ConfigException {
         if (!VcHelper.filterVCBasedOnProfileType(vcStoreItems, ProfileType.GPG45).isEmpty()) {
             LOGGER.info(
                     LogHelper.buildLogMessage("Failed to match profile so resetting identity."));
             sendAuditEvent(AuditEventTypes.IPV_IDENTITY_REUSE_RESET, auditEventUser);
+            if (ciMitUtilityService.hasMitigatedContraIndicator(contraIndicators).isPresent()) {
+                return ciMitUtilityService
+                        .getCiMitigationJourneyStep(contraIndicators)
+                        .orElse(JOURNEY_FAIL_WITH_CI);
+            }
```

Comment:
I think this might need to be outside of this `if` block. That `if` is there to ensure we reset the users VC before sending them on a full IPV journey, if we need to.

The separate session mitigation routes will all reset the users VC anyway, so we don't need to worry about that.

In this `buildNoMatchResponse` method I think we want to do three things:
1. Check if the user has mitigated CI already - if they do find the mitigation journey step from CI config for that CI and send them on it. If they don't have a mitigated CI, continue.
2. Check if they've got any GPG45 VC. If they do, send them on `JOURNEY_RESET_GPG45_IDENTITY`. If they don't, continue.
3. If we've not returned already, return `JOURNEY_IPV_GPG45_MEDIUM`.

We're already doing step 2 and 3, so you should be able to just slide in above them.

I don't think your use of `getCiMitigationJourneyStep` will give you what you want though. That method will only look at unmitigated CI. Maybe you need a new method in the `cimitUtilitySerice` - something like `getMitigatedCiJourneyStep` which takes a single CI. You could pass it the result of `hasMitigatedContraIndicator` which returns an optional of the mitigated CI.